src/rewards/rewards.service.ts

```diff
@@ -41,4 +46,50 @@ export class RewardsService {
     }
     return claim;
   }
+
+  /**
+   * Get all token rewards emmited, by vault
+   * @param chain Network chain obj
+   * @param address Vault token adress
+   */
+  async rewardSchedulesByVault(chain: Chain, address: VaultDefinition['vaultToken']): Promise<RewardSchedulesByVault> {
+    const rewardsLogger = getRewordLoggerInst(chain);
+
+    if (!rewardsLogger) throw new NotFound(`No rewards token emission data for network ${Chain.name}`);
+
+    const vault = chain.vaults.find((v) => v.vaultToken === address);
+
+    if (!vault || !vault?.vaultToken) throw new NotFound(`Unknown vault ${address}`);
+
+    return {
+      schedules: await getRewardSchedules(chain, vault.vaultToken, rewardsLogger),
```

Comment:
there is no need to search for vault because you are passing a vault definition and the vault token in so you already have what you are looking for from the input

Reply:
ye, just checking if vault exists